Tests/BootCamp.Chapter.Tests/TestsDeck.cs

```diff
@@ -0,0 +1,144 @@
+﻿using BootCamp.Chapter.Gambling;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace BootCamp.Chapter.Tests
+{
+    public class TestsDeck
+    {
+        private static List<Card> FullDeckOfCards()
+        {
+            List<Card> cardsDeck = new List<Card>();
+            foreach (Card.Suites suite in (Card.Suites[])Enum.GetValues(typeof(Card.Suites)))
+            {
+                foreach (Card.Ranks rank in (Card.Ranks[])Enum.GetValues(typeof(Card.Ranks)))
+                {
+                    cardsDeck.Add(new Card(suite, rank));
+                }
+            }
+            return cardsDeck;
+        }
+
+        [Fact]
+        public void When_No_Cards_Given_To_Deck_Should_Throw_ArgumentNullException()
```

Comment:
All tests should start with the action name, followed by preconditions and parameters, followed by expectation.
``Act_Given_Precondition_When_Argument_Expectation``
In your case, it should be: 
`New_When_CardsNull_Throws_ArgumentException`

Reply:
much more readable.